presto-orc/src/main/java/com/facebook/presto/orc/metadata/statistics/IntegerStatisticsBuilder.java

```diff
@@ -33,6 +36,13 @@ public void addValue(long value)
 
         minimum = Math.min(value, minimum);
         maximum = Math.max(value, maximum);
+
+        try {
```

Comment:
I would only execute this try if `overflow` is not already set. Given the way this code is currently written, I'm concerned about repeated throws in a tight loop.